### PR TITLE
Upgrade OpenSSL from `1.1.1` to `3.1.3`

### LINKED ISSUE #4159: Static build CKB's release with `openssl 3`
#### Is your feature request related to a problem? Please describe.

The OpenSSL 1.1.1 series has reached its [End of Life](https://www.openssl.org/blog/blog/2023/09/11/eol-111), so it's time for us to transition to OpenSSL 3 since CKB is statically built with OpenSSL 1.1.1.

https://github.com/nervosnetwork/ckb/blob/6c6ea88c7605aaed1f955485205745dd9cb0fa0c/.github/workflows/package.yaml#L113-L115

https://github.com/nervosnetwork/ckb/blob/6c6ea88c7605aaed1f955485205745dd9cb0fa0c/.github/workflows/package.yaml#L198

https://github.com/nervosnetwork/ckb/blob/6c6ea88c7605aaed1f955485205745dd9cb0fa0c/.github/workflows/package.yaml#L257

We should also consider upgrading the `ckb-docker-builder`:

https://github.com/nervosnetwork/ckb-docker-builder/blob/8ce6f4d4a073be833ef07f6a61d038eeb12b5780/centos-7/Dockerfile#L11-L12

### PR DESCRIPTION
**Pull request by eval-exec**
### What problem does this PR solve?

Issue Number: close #4159 

Problem Summary:
The OpenSSL 1.1.1 series has reached its [End of Life](https://www.openssl.org/blog/blog/2023/09/11/eol-111), so it's time for us to transition to OpenSSL 3 since CKB is statically built with OpenSSL 1.1.1.
### What is changed and how it works?

What's Changed:

### Related changes

- [x] staticly linked openssl 3.1.3 on `Docker` image build
- [x] staticly linked openssl 3.1.3 on `Linux` platform release
- [x] staticly linked openssl 3.1.3 on `Macos` platform release
- [ ] ~staticly linked openssl 3.1.3 on `Windows` platform release~
- [x] test `package.yaml` CI workflow on all platform
- [x] test staticly linked openssl 3.1.3 docker image

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code ci-runs-only: [ quick_checks,linters ]

Side effects

- None

### Release note 

```release-note
Title Only: Include only the PR title in the release note.
```

### CODE REVIEW

**Comment by eval-exec** [comment]
This PR need https://github.com/nervosnetwork/ckb-docker-builder/pull/17

**Comment by doitian** [comment]
docker builder 1.71.0-openssl-3.1.3 published

**Comment by eval-exec** [comment]
> docker builder 1.71.0-openssl-3.1.3 published

The docker image's tag should be `1.71.1-openssl-3.1.3` instead of `1.71.0-openssl-3.1.3`. Can we revoke `nervos/ckb-docker-builder:1.71.0-openssl-3.1.3`?

**Comment by eval-exec** [comment]
Waiting https://github.com/nervosnetwork/ckb-docker-builder/issues/19

**Comment by eval-exec** [comment]
On x86_64-darwin64, `make prod` got this:
```
bash-3.2$ lldb ./target/prod/ckb
(lldb) target create "./target/prod/ckb"
Current executable set to '/Users/runner/work/ckb/ckb/target/prod/ckb' (x86_64).
(lldb) b main
Breakpoint 1: 10 locations.
(lldb) r
Process 37067 launched: '/Users/runner/work/ckb/ckb/target/prod/ckb' (x86_64)
Process 37067 stopped
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_INSTRUCTION (code=EXC_I386_INVOP, subcode=0x0)
    frame #0: 0x000000010133321d ckb`std::__1::unordered_map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, rocksdb::OptionTypeInfo, std::__1::hash<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, std::__1::equal_to<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const, rocksdb::OptionTypeInfo> > >::unordered_map(std::initializer_list<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const, rocksdb::OptionTypeInfo> >) + 1005
ckb`std::__1::unordered_map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, rocksdb::OptionTypeInfo, std::__1::hash<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, std::__1::equal_to<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const, rocksdb::OptionTypeInfo> > >::unordered_map:
->  0x10133321d <+1005>: blsrq  %rdi, %rcx
    0x101333222 <+1010>: setne  %cl
    0x101333225 <+1013>: orb    %al, %cl
    0x101333227 <+1015>: vdivss %xmm1, %xmm0, %xmm0
Target 0: (ckb) stopped.
```

File: .github/workflows/package.yaml [hunk 1]
```diff
@@ -78,7 +78,7 @@ jobs:
         name: ckb_${{env.GIT_TAG_NAME }}_${{env.REL_PKG }}.asc
         path: ckb_${{env.GIT_TAG_NAME }}_${{env.REL_PKG }}.asc
     env:
-      BUILDER_IMAGE: nervos/ckb-docker-builder:bionic-rust-1.71.1
+      BUILDER_IMAGE: nervos/ckb-docker-builder:bionic-rust-1.71.1-openssl-3.1.3
```
Comment (eval-exec) [inline .github/workflows/package.yaml:81]: `nervos/ckb-docker-builder:bionic-rust-1.71.1-openssl-3.1.3` doesn't exist in dockerhub.
In dockerhub, we have `[bionic-rust-1.71.0-openssl-3.1.3](https://hub.docker.com/layers/nervos/ckb-docker-builder/bionic-rust-1.71.0-openssl-3.1.3/images/sha256-006f771d3493ad1508fdea1c49c0522129bc5d1e264a8149e20bedca20911654?context=explore)` and [centos-7-rust-1.71.0-openssl-3.1.3](https://hub.docker.com/layers/nervos/ckb-docker-builder/centos-7-rust-1.71.0-openssl-3.1.3/images/sha256-5c0542482215b334c03cc261a1eacb4b6bd28dff70bee0f322b60b52a3452670?context=explore)

**Comment by eval-exec** [comment]
This PR is ready to be merged.

**Review by doitian — APPROVED** [review]

**Comment by eval-exec** [comment]
The CI job on Ubuntu ([link](https://github.com/nervosnetwork/ckb/actions/runs/7166427065/job/19510431339)) failed without an error log. I have re-triggered the failed jobs.

**Comment by eval-exec** [comment]
@doitian I think 0.113.0 need this PR.

**Pull request merged**